subsys/bluetooth/controller/hal/nrf5/radio/radio.c

```diff
@@ -791,22 +800,36 @@ void radio_tmr_end_capture(void)
 		HAL_RADIO_END_TIME_CAPTURE_PPI_EVT;
 	HAL_RADIO_END_TIME_CAPTURE_PPI_REGISTER_TASK =
 		HAL_RADIO_END_TIME_CAPTURE_PPI_TASK;
-	NRF_PPI->CHENSET = HAL_RADIO_END_TIME_CAPTURE_PPI_ENABLE;
+
+	HAL_EVENT_TIMER_CLEAR_PPI_REGISTER_EVT = HAL_EVENT_TIMER_CLEAR_PPI_EVT;
```

Comment:
My origin code before refactor used SOC registers directly for the reason I dont have to remember or read some other header file to know what is the CLEAR_PPI_EVT ?... ok... after some greping its the NRF_RADIO->EVENTS_END;

So now you are clearing a free running timer, but keeping track of the microseconds in a local variable. Clearing the timer would add a +/- 1 us drift?

Reply:
Clearing the timer, actually, removes jitter/drift for HCTO, PA/LNA and TIFS calculations.
It, however, introduces a drift of the conception of the absolute time inside the event. I wonder whether the absolute time is used anywhere.